### ISSUE TITLE
how to evaluate a model?

### ISSUE THREAD

**Issue by zhifeng-huang** [tensorflow/recommenders#249]
I have followed the tutorial to implement a model for my own dataset. However, I have no idea how good the performance it is. 

I am turning hyperparameter. 

What does it mean if I get the following result?
Top-100 accuracy (train): 0.28.
Top-100 accuracy (test): 0.09.

What does it mean if I get the following result?
Top-100 accuracy (train): 0.22.
Top-100 accuracy (test): 0.22.

I trained the model so the loss did not go down anymore. I still could not bring up the accuracy. Any suggestion what I can do?

**Comment by turnerneil** [comment]
You first results indicate the model is overfitting. This means it wont generalize well. It's simply memorizing the training data. 

I'de recommend reading "[Examples of Bias and Variance](https://gallery.mailchimp.com/dc3a7ef4d750c0abfc19202a3/files/5dd91615-3b3f-4f5d-bbfb-4ebd8608d330/Ng_MLY01_13.pdf)" by Andrew Ng for a basic overview of what these numbers mean.

**Comment by dgoldenberg-audiomack** [comment]
I think the doc would benefit from a specific how-to on how to tell if it's overfitting and what to try to combat the issue.

**Comment by almirb** [comment]
> I think the doc would benefit from a specific how-to on how to tell if it's overfitting and what to try to combat the issue.

I think this would clarify overfit a little:
https://www.tensorflow.org/tutorials/keras/overfit_and_underfit

**Comment by dgoldenberg-audiomack** [comment]
@almirb Yeah, nice but won't this take months or years to get all done and right? I think there are mlops frameworks out there to automate this kind of thing, otherwise it's an endless trial-and-error loop...

**Comment by almirb** [comment]
> @almirb Yeah, nice but won't this take months or years to get all done and right? I think there are mlops frameworks out there to automate this kind of thing, otherwise it's an endless trial-and-error loop...

Yes.. I am also  getting a hard time trying to get all those things to work and create a production grade recsys. 
TFRS is a great framework! But I'm new to keras and tensorflow.  A lot of things to learn...
